### DISCUSSION TITLE
How to reduce initial loading time?

### DISCUSSION THREAD

**Discussion by DavidBerger98** [Kitware/trame#151] [Q&A]
When the client connects to the trame server, there is a loading screen. I was asking myself if there's a way to make it faster or whether there are things I should be mindful of to not make it slower.

I am guessing this is when trame builds the html and js code and sends the template to the client. Does trame execute the Python code each time a client connects? If yes, is there a way we could tell trame to "compile" the template on startup and then just serve it like a classic http server?

Futhermore, is there a way to "off-load" some of the loading time to a later point in the application, i.e. have a "static" landing page that loads pretty much instantly and have the user then connect to the trame application?

I know there's a way to add custom paths to the trame server like this:

```py
from aiohttp import web
from trame.app import get_server

server = get_server()

async def hello_world(request):
    return web.Response(text="Hello world")

@server.controller.add("on_server_bind")
def app_available(wslink_server):
    wslink_server.app.add_routes([web.get("/hello", hello_world)])
```

But what is your recommended way to communicate between the static part and trame? URL parameters?

**Comment by jourdain** [comment]
The server is already waiting and idle when the client connects. Unless the client tries to reach out to the server before the server is listening/ready.

The initial wait time should just be loading the server state (that should already be computed) which could be big? Then the client loads all the JS and CSS it needs and display the compiled template that was provided at the initial state exchange. 

How much of a wait are we talking about? Maybe you could update/compute the state only once the client connect (kind of faking some data streaming).

For serving static content, you can create a module like any of the [trame-xxx library](https://github.com/Kitware/trame-vtk/blob/master/trame_vtk/modules/common/__init__.py#L4-L7) and load/enable it. 

For the last question, I'm not sure I understand the question.
